packages/Kokkos/test/tstLinearBVH.cpp

```diff
@@ -295,87 +356,109 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, rtree, NO )
         rtree.query( bgi::satisfies( [centroid, radius](
-                         std::pair<Point, int> const &val ) {
+                         std::pair<BPoint, int> const &val ) {
                          return bg::distance( centroid, val.first ) <= radius;
                      } ),
-                     std::back_inserter( returned_values ) );
-
-        // I tried to encapsulate this in a lambda but it won't work with the
-        // Teuchos test assertion macros
-        // Anyway, this is fine for now...
-        {
-            auto const &ref = returned_values;
-            auto compare = []( std::pair<int, double> const &a,
-                               std::pair<int, double> const &b ) {
-                return a.first < b.first;
-            };
-            std::set<std::pair<int, double>, decltype( compare )> tmp(
-                compare );
-            tmp.insert( sol.begin(), sol.end() );
-            TEST_EQUALITY( sol.size(), ref.size() );
-            for ( auto const &x : ref )
-            {
-                auto it = tmp.find( std::make_pair( x.second, -1.0 ) );
-                TEST_ASSERT( it != tmp.end() );
-                TEST_EQUALITY( it->second, bg::distance( centroid, x.first ) );
-            }
-        }
+                     std::back_inserter( returned_values_within[i] ) );
 
         // k nearest neighbors
-        sol.clear();
-        returned_values.clear();
+        rtree.query( bgi::nearest( BPoint( x, y, z ), k[i] ),
+                     std::back_inserter( returned_values_nearest[i] ) );
+    }
 
-        rtree.query( bgi::nearest( Point( x, y, z ), k ),
-                     std::back_inserter( returned_values ) );
-        sol = DataTransferKit::Details::nearest( bvh, {x, y, z}, k );
-        // copy/paste of the check solution against reference above
-        {
-            auto const &ref = returned_values;
-            auto compare = []( std::pair<int, double> const &a,
-                               std::pair<int, double> const &b ) {
-                return a.first < b.first;
-            };
-            std::set<std::pair<int, double>, decltype( compare )> tmp(
-                compare );
-            tmp.insert( sol.begin(), sol.end() );
-            TEST_EQUALITY( sol.size(), ref.size() );
-            for ( auto const &x : ref )
-            {
-                auto it = tmp.find( std::make_pair( x.second, -1.0 ) );
-                TEST_ASSERT( it != tmp.end() );
-                TEST_EQUALITY( it->second, bg::distance( centroid, x.first ) );
-            }
-        }
+    auto random_within_lambda = KOKKOS_LAMBDA( const int i )
+    {
+        details::Within within_predicate(
+            {point_coords( i, 0 ), point_coords( i, 1 ), point_coords( i, 2 )},
+            radii( i ) );
+        auto sol_within = details::spatial_query( bvh, within_predicate );
+        within_n_pts( i, 0 ) = sol_within.size();
+        within_n_pts( i, 1 ) = *sol_within.begin();
```

Comment:
wait a min, so we only store the 1st one that was found (provided that there were collisions)??